### PR TITLE
database: optional pprof for tenantless_queries

### PR DESCRIPTION
**Pull request by keegancsmith**
This adds a new profile output for when a sql query does not have a tenant set. This can be downloaded like other pprof profiles and inspected by "go tool pprof". This allows us to easily see the most common stack traces and sources of missing context.

Test Plan: downloaded from the debugserver the new profile type and inspected with "go tool pprof"

### CODE REVIEW

File: internal/database/dbconn/tenant_pprof.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	if _, ok := tenant.FromContext(ctx); ok {
+		return
+	}
+
+	// Allowed tenantless query
+	if strings.Contains(query, "critical_and_site_config.id") {
+		return
+	}
```
Comment (eseliger) [inline internal/database/dbconn/tenant_pprof.go:34]: This was just to reduce logspam locally so I could see anything useful at all, I think we'd be okay not skipping it here to keep it more realistic, wdyt?

**Review by eseliger — APPROVED** [review]
might need a `sg bazel configure`

File: internal/database/dbconn/tenant_pprof.go (hunk 1 above)
Comment (keegancsmith) [reply internal/database/dbconn/tenant_pprof.go:34]: ok will update

File: internal/database/dbconn/tenant_pprof.go [hunk 2]
```diff
@@ -19,7 +18,7 @@ var pprofTenantlessQueries = func() *pprof.Profile {
 	return pprof.NewProfile("tenantless_queries")
 }()
 
-func pprofCheckTenantlessQuery(ctx context.Context, query string) {
+func pprofCheckTenantlessQuery(ctx context.Context) {
 	if pprofTenantlessQueries == nil {
```
Comment (eseliger) [inline internal/database/dbconn/tenant_pprof.go:22]: we should use SRC_TENANT_ENFORCEMENT_MODE here now :)

**Pull request merged**